utils/Azure/acrTools.ts

```diff
@@ -57,57 +58,98 @@ export async function getResourceGroup(registry: Registry, subscription: Subscri
+    let response = await sendRequest<{ repositories: string[] }>('get', getLoginServer(registry), 'acr/v1/_catalog', acrAccessToken);
 
     let allRepos: Repository[] = [];
-    for (let tempRepo of repositories) {
-        allRepos.push(await Repository.Create(registry, tempRepo));
+    if (!isNull(response.repositories)) {
+        for (let tempRepo of response.repositories) {
+            allRepos.push(await Repository.Create(registry, tempRepo));
+        }
     }
+
     //Note these are ordered by default in alphabetical order
     return allRepos;
 }
 
-/** List images under a specific digest */
-export async function getImagesByDigest(repo: Repository, digest: string): Promise<AzureImage[]> {
-    let allImages: AzureImage[] = [];
-    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(repo.registry, 'repository:' + repo.name + ':pull');
-    const tags: string[] = await getAssociatedTags('https://' + repo.registry.loginServer, repo.name, digest, { bearer: acrAccessToken });
-    for (let tag of tags) {
-        allImages.push(new AzureImage(repo, tag));
-    }
-    return allImages;
+export async function deleteRepository(repo: Repository): Promise<void> {
+    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(repo.registry, `repository:${repo.name}:*`);
+    await sendRequest('delete', getLoginServer(repo.registry), `v2/_acr/${repo.name}/repository`, acrAccessToken);
+}
+
+export async function deleteImage(repo: Repository, imageDigest: string): Promise<void> {
+    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(repo.registry, `repository:${repo.name}:*`);
+    await sendRequest('delete', getLoginServer(repo.registry), `v2/${repo.name}/manifests/${imageDigest}`, acrAccessToken);
+}
+
+export async function untagImage(img: AzureImage): Promise<void> {
+    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(img.registry, `repository:${img.repository.name}:*`);
+    await sendRequest('delete', getLoginServer(img.registry), `v2/_acr/${img.repository.name}/tags/${img.tag}`, acrAccessToken);
 }
 
 /** Sends a custom html request to a registry
- * @param http_method : the http method, this function currently only uses delete
+ * @param http_method : the http method
  * @param login_server: the login server of the registry
  * @param path : the URL path
  * @param accessToken : Bearer access token.
  */
-export async function sendRequestToRegistry(http_method: 'delete', login_server: string, path: string, accessToken: string): Promise<void> {
-    let url: string = `https://${login_server}${path}`;
-    let opt = {
-        headers: { 'Authorization': `Bearer ${accessToken}` },
-        http_method: http_method,
-        url: url
-    }
+export async function sendRequest<T>(http_method: string, login_server: string, path: string, accessToken: string): Promise<T> {
+    let url: string = `https://${login_server}/${path}`;
```

Comment:
remove strictSSL and related code (and don't pass anything in to request.get, let it use the default).